### PR TITLE
Plugin connectivity improvements

### PR DESCRIPTION
**Pull request by neiljerram**
This PR also adds unit testing of the Neutron driver that raises its code coverage to 57%.

### CODE REVIEW

**Comment by Lukasa** [comment]
LGTM. :cake:

**Comment by neiljerram** [comment]
Could I merge this now, or are there considerations against doing that?

**Comment by Lukasa** [comment]
It's not clear to me that I should be code reviewing this. =) Recommend that @plwhite takes a look too.

**Comment by neiljerram** [comment]
Thanks @plwhite !

File: calico/openstack/mech_calico.py [hunk 1]
```diff
@@ -607,6 +621,11 @@ def felix_heartbeat_thread(self, hostname):
                 else:
                     LOG.error("Unexpected response from Felix on %s: %s"
                               % (hostname, rsp))
+            except Again:
+                LOG.error("No response from Felix within allowed time (%sms)" %
+                          HEARTBEAT_RESPONSE_TIMEOUT)
```
Comment (plwhite) [inline calico/openstack/mech_calico.py]: Shouldn't you throw a FelixUnavailable exception here, as in the comment above?

Comment (neiljerram) [inline calico/openstack/mech_calico.py]: No, because there is no caller who'd be interested in that - we're on a dedicated heartbeat thread here.

Comment (plwhite) [inline calico/openstack/mech_calico.py]: The comment not so many lines above says that you do raise the exception, and I think should be tidied up (presumably a bad block copy). Then I'm happy to merge.

Comment (neiljerram) [inline calico/openstack/mech_calico.py]: I think the comment is meaning that the poll call will raise an exception - but I'm happy to clarify that, if that will address your concern here.

Comment (neiljerram) [inline calico/openstack/mech_calico.py]: Sorry, I mean the recv_json call.

**Pull request merged**